packages/shared/src/components/squads/SquadMemberRoleBadge.tsx

```diff
@@ -0,0 +1,39 @@
+import React, { ReactElement } from 'react';
+import { Author } from '../../graphql/comments';
+import { Source, SourceMemberRole } from '../../graphql/sources';
+import { useMemberRoleForSource } from '../../hooks/useMemberRoleForSource';
+import StarIcon from '../icons/Star';
+import UserIcon from '../icons/User';
+import UserBadge from '../UserBadge';
+
+export type SquadMemberRoleBadgeProps = {
+  source?: Source;
+  author?: Author;
+};
+
+const RoleToIconMap = {
+  [SourceMemberRole.Moderator]: UserIcon,
+  [SourceMemberRole.Owner]: StarIcon,
+};
+
+const SquadMemberRoleBadge = ({
+  author,
+  source,
+}: SquadMemberRoleBadgeProps): ReactElement => {
+  const { role } = useMemberRoleForSource({ source, user: author });
+  const RoleIcon = RoleToIconMap[role];
+
+  if (!RoleIcon) {
+    return null;
+  }
```

Comment:
Still left this mapping logic to component level, I don't think its a good pattern to import and returns icons/components from hooks. View related stuff is best left in components.

Reply:
Totally agree with that. No point in importing something you can't render. Basically in a hook. Well done.